src/payment/strategies/zip/zip-payment-strategy.spec.ts

```diff
@@ -0,0 +1,271 @@
+
+        strategy = new ZipPaymentStrategy(
+            store,
+            orderActionCreator,
+            paymentActionCreator,
+            paymentMethodActionCreator,
+            zipScriptLoader
+        );
+
+        jest.spyOn(paymentMethodActionCreator, 'loadPaymentMethod');
+    });
+
+    it('creates an instance of ZipPaymentStrategy', () => {
+        expect(strategy).toBeInstanceOf(ZipPaymentStrategy);
+    });
+
+    describe('#initialize()', () => {
+        let zipOptions: PaymentInitializeOptions;
+
+        beforeEach(() => {
+            zipOptions = { methodId: 'zip' };
+        });
+
+        it('initializes the strategy successfully', async () => {
+            await strategy.initialize(zipOptions);
+
+            expect(paymentMethodActionCreator.loadPaymentMethod).toHaveBeenCalledWith(zipOptions.methodId);
+            expect(zipScriptLoader.load).toHaveBeenCalled();
+        });
+
+        it('does not initialize the strategy if method id is not provided', async () => {
+            zipOptions.methodId = '';
+            await expect(() => strategy.initialize(zipOptions)).toThrowError(NotInitializedError);
+            expect(zipScriptLoader.load).not.toHaveBeenCalled();
+        });
+
+        it('does not initialize the strategy if payment method is not provided', async () => {
+            jest.spyOn(store.getState().paymentMethods, 'getPaymentMethod')
+                .mockReturnValueOnce(undefined);
+            try {
+                await strategy.initialize(zipOptions);
+            } catch (error) {
+                expect(error).toBeInstanceOf(MissingDataError);
+                expect(zipScriptLoader.load).not.toHaveBeenCalled();
+            }
+        });
+
+        it('does not initialize the strategy if client token is not provided', async () => {
+            paymentMethodMock.clientToken = undefined;
+            jest.spyOn(store.getState().paymentMethods, 'getPaymentMethod')
+                .mockReturnValue(paymentMethodMock);
+            try {
+                await strategy.initialize(zipOptions);
+            } catch (error) {
+                expect(error).toBeInstanceOf(MissingDataError);
+                expect(zipScriptLoader.load).not.toHaveBeenCalled();
+            }
+        });
+    });
+
+    describe('#execute()', () => {
+        let zipOptions: PaymentInitializeOptions;
+        let orderRequestBody: OrderRequestBody;
+        const zipInitPayload =  {
+                    onComplete: expect.any(Function),
+                    onCheckout: expect.any(Function),
+                };
+
+        beforeEach(async () => {
+            orderRequestBody = getOrderRequestBody();
+            zipOptions = { methodId: 'zip' };
+
+            await strategy.initialize(zipOptions);
+        });
+
+        it('executes the strategy successfully and submits the payment', async () => {
+            const { payment, ...order } = orderRequestBody;
+            const expected = {
```

Comment:
`expected` what? We're asserting several things here so it's good to be specific about what you're expecting.